cmd/icingadb/main.go

```diff
@@ -33,47 +35,76 @@ func main() {
 
 func run() int {
 	cmd := command.New()
+	// When started by systemd, NOTIFY_SOCKET is set by systemd for Type=notify supervised services, which is the
+	// default setting for the Icinga DB service. So we notify that Icinga DB finished starting up and also configure
+	// logging into the systemd journal instead of stderr.
+	// Always add this line before output assignment
+	_, _ = daemon.SdNotify(false, daemon.SdNotifyReady)
+	var output string
+	// When the type in icingadb.service is set to notify then set output to "systemd-journal",
+	// else set it to console, that is write log messages to stdErr.
+	if _, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
+		output = "systemd-journal"
```

Comment:
Yes, `Type = notify` is only there to say if it runs under systemd. I would leave it that way, because the alternative can write to the journal if systemd is available and Icinga DB is started somehow differently.

Reply:
Well, what about just doing `ExecStart=/usr/bin/icingadb -log journald`? Would make it obvious why it logs to which destination.